ceno_zkvm/src/instructions/riscv/shift_imm.rs

```diff
@@ -181,6 +181,12 @@ impl<E: ExtensionField, I: RIVInstruction> Instruction<E> for ShiftImmInstructio
     }
 }
 
+/// Decode the immediate into the format of this circuit.
+/// The shift is implemented as a multiplication/division by 1 << immediate.
+pub fn process_imm(insn: &DecodedInstruction) -> u32 {
```

Comment:
I would like to propose to move this function to `program.rs`. It would be easier for circuit writers, they don't need to care the bits shifting. What they need to do is just calling `process_imm` and `process_imm` will return what they expected.

Reply:
We should just record the proper shift amount in a field named 'immediate' in the Decoded instruction.

Reply:
Done.